src/Magnum/Math/StrictWeakOrdering.h

```diff
@@ -0,0 +1,93 @@
+              Vladimír Vondruš <mosra@centrum.cz>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ref Magnum::Math::StrictWeakOrdering
+ */
+
+namespace Magnum { namespace Math {
+/**
+@brief A functor which implements strict weak ordering for types in the math
+library.
+@param T the type for which to have strict weak ordering.
+
+A strict weak ordering enables a comparison operator between to elements of a
+set. With it an ordering can be achieved between elements.
+
+This is useful mainly for interoperability with ordering containers from the
+C++ standard library, like std::map and std::set. Instead of using this
+functor, it is recommended you use @ref StrictWeakOrderingFunc which provides
+a shorthand interface to this type.
+
+If you want to enable strict weak ordering through this or the
+@ref StrictWeakOrderingFunc shorthand, you must specialize this struct for your
+own type.
+ */
+template <class T> // Specialized for supported types
+struct StrictWeakOrdering {
+    /**
+     * @brief Compares two items
+     *
+     * Performs the comparison. For scalar types this is equivalent to
+     * operator <. For otther types specializations of this struct exist which
+     * have a specific implementation for the comparison such that is follows
+     * the definition of strict weak ordering.
+     */
+    bool operator()(const T& a, const T& b) const {
+        return a < b;
+    }
+};
+
+/**
+@brief A shorthand interface for @ref StrictWeakOrdering
+
+Allows the use of @ref StrictWeakOrdering in a shorter and more succinct way.
+To use for std::map or std::set, just add it as their compare template
+argument.
+
+@code{.cpp}
+std::set<Magnum::Vector2, Magnum::Math::StrictWeakOrderingFunc> mySet;
+std::map<Magnum::Vector4, int, Magnum::Math::StrictWeakOrderingFunc> myMap;
+@endcode
+ */
+struct StrictWeakOrderingFunc {
```

Comment:
An idea regarding the naming: since `StrictWeakOrderingFunc` will be used 99% of the time and people wouldn't even need to know that `StrictWeakOrdering` exists, what about reducing the API surface area a bit:

- move `StrictWeakOrdering` struct into the `Math::Implementation` namespace (it's hidden from Doxygen, so then you also don't need to have the `#ifndef`s around each)
- rename `StrictWeakOrderingFunc` to `StrictWeakOrdering` and merge documentation of these two, so instead of:

    ```
    If you want to enable strict weak ordering through this or the
    @ref StrictWeakOrderingFunc shorthand, you must specialize this struct for your
    own type.
    ```

    just mention that the this is implemented for all `Math` types and handwave the implementation by mentioning the struct specialization in `Math::Implementation` and advising the users to check source for details (... but I just realized it's not done for `Bezier` and `CubicHermite`, can you add a specialization for these as well?)
- the tests could keep using the `StrictWeakOrdering` (of course without the explicit template param after) to verify the shorthand also does the job correctly

Reply:
Sure. I'll do it tonight. 

I hadn't actually realized that Bezier and CubicHermite were types. I thought they were just files with functions for splines. 😄

Reply:
Oh and `Range` as well. Then there's `Half`, which is a completely opaque type for 16-bit floats (and thus no `operator<`). Not sure about that one.

The `Deg`, `Rad` types (derived from `Unit`) have `operator<` implemented so nothing needed I think.

Reply:
Ah, so `Half` doesn't have an `operator<`. I'll create an specialization for it as well then. It makes sense to have a way to order them.